internal/test/builder/builders.go

```diff
@@ -1069,7 +1330,7 @@ func (m *MachinePoolBuilder) WithBootstrapTemplate(ref *unstructured.Unstructure
 	return m
 }
 
-// WithInfrastructureTemplate adds the passed unstructured object to the MachinePool builder as an infrastructureMachineTemplate.
+// WithInfrastructureTemplate adds the passed unstructured object to the MachinePool builder as an infrastructureMachinePoolTemplate.
```

Comment:
```suggestion
// WithInfrastructure adds the passed unstructured object to the MachinePool builder as an InfrastructureMachinePool.
```
this is not a template here